components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/auth/OAuthHandler.java

```diff
@@ -184,6 +189,29 @@ private OAuth2ClientApplicationDTO validateAccessToken(String accessTokenIdentif
             OAuth2ClientApplicationDTO oauthValidationResponse = oauthValidationService
                     .findOAuthConsumerIfTokenIsValid(oauthValidationRequest);
 
+            String restrictFederatedUserAccess =
+                    IdentityUtil.getProperty(SCIMCommonConstants.SCIM_RESTRICT_FEDERATED_USER_ACCESS_TO_ME_ENDPOINT);
+            if (Boolean.parseBoolean(restrictFederatedUserAccess)) {
+                OAuth2IntrospectionResponseDTO oAuth2IntrospectionResponseDTO =
```

Comment:
Check whether we can do a mapping between oAuth2IntrospectionResponseDTO and oauthValidationResponse when fixing it in master

Reply:
Shall we improve the logic to only do the mapping between oAuth2IntrospectionResponseDTO and oauthValidationResponse if the config is enabled?